cudamapper/include/claragenomics/cudamapper/types.hpp

```diff
@@ -89,21 +79,21 @@ typedef struct Overlap
     /// end position in the target
     position_in_read_t target_end_position_in_read_;
     /// query read name (e.g from FASTA)
-    char* query_read_name_ = 0;
+    char* query_read_name_;
```

Comment:
@vellamike @mimaric Are we sure the code doesn't assume the initialization anywhere?

Reply:
This depends on #281. This whole part of code is rather messy because it uses raw pointers. @vellamike and I tried to make it nicer using smart pointers, but we were either running into performance issues or thurst was refusing to compile. We could probably fix it using preprocessor macros, but for now we left it as it is.
Our clear function (added in #281, this version actually has a memory leak) relies on this being set to null when there is not value, but it's definitely not the best design ever, we just didn't have time to dive deeper into to it and find a performant solution which also work with thrust.
What's the reason for having this not default initialized?

Reply:
The problem is that @akkamesh's buffer only supports trivial types (i.e. plain old data structures), because it doesn't call any destructors on shrinking or destruction - it just deallocates memory. It does not call constructors either, but this is by design.

Reply:
Ah I think we can relax this. The Trivial Copyable concept should be good. I just don't want the user to assume that destructors are called.
@akkamesh could you undo this change here? I.e. leave all the default initializers in Overlap?

Then in the device buffer, please change the
```
static_assert(std::is_trivial<value_type>::value, "buffer only supports trivial types and classes.");
```
to
```
static_assert(std::is_trivially_copyable<value_type>::value, "buffer only supports trivially copyable types and classes, because destructors will not be called.");
```

Reply:
done